python/private/pypi/whl_library_targets.bzl

```diff
@@ -0,0 +1,339 @@
+            # Ensure that this works on Windows as well - script may have Windows path separators.
+            srcs = [entry_point_script_name.replace("\\", "/")],
+            # This makes this directory a top-level in the python import
+            # search path for anything that depends on this.
+            imports = ["."],
+            deps = [":" + PY_LIBRARY_PUBLIC_LABEL],
+            visibility = ["//visibility:public"],
+        )
+
+    # Ensure this list is normalized
+    # Note: mapping used as set
+    group_deps = {
+        normalize_name(d): True
+        for d in group_deps
+    }
+
+    dependencies = [
+        d
+        for d in dependencies
+        if d not in group_deps
+    ]
+    dependencies_by_platform = {
+        p: deps
+        for p, deps in dependencies_by_platform.items()
+        for deps in [[d for d in deps if d not in group_deps]]
+        if deps
+    }
+
+    # If this library is a member of a group, its public label aliases need to
+    # point to the group implementation rule not the implementation rules. We
+    # also need to mark the implementation rules as visible to the group
+    # implementation.
+    if group_name and "//:" in dep_template:
+        # This is the legacy behaviour where the group library is outside the hub repo
+        label_tmpl = dep_template.format(
+            name = "_groups",
+            target = normalize_name(group_name) + "_{}",
+        )
+        impl_vis = [dep_template.format(
+            name = "_groups",
+            target = "__pkg__",
+        )]
+
+        native.alias(
+            name = PY_LIBRARY_PUBLIC_LABEL,
+            actual = label_tmpl.format(PY_LIBRARY_PUBLIC_LABEL),
+            visibility = ["//visibility:public"],
+        )
+        native.alias(
+            name = WHEEL_FILE_PUBLIC_LABEL,
+            actual = label_tmpl.format(WHEEL_FILE_PUBLIC_LABEL),
+            visibility = ["//visibility:public"],
+        )
+        py_library_label = PY_LIBRARY_IMPL_LABEL
+        whl_file_label = WHEEL_FILE_IMPL_LABEL
+
+    elif group_name:
+        py_library_label = PY_LIBRARY_PUBLIC_LABEL
+        whl_file_label = WHEEL_FILE_PUBLIC_LABEL
+        impl_vis = [dep_template.format(name = "", target = "__subpackages__")]
+
+    else:
+        py_library_label = PY_LIBRARY_PUBLIC_LABEL
+        whl_file_label = WHEEL_FILE_PUBLIC_LABEL
+        impl_vis = ["//visibility:public"]
+
+    if hasattr(native, "filegroup"):
+        native.filegroup(
+            name = whl_file_label,
+            srcs = [whl_name],
+            data = _deps(
+                deps = dependencies,
+                deps_by_platform = dependencies_by_platform,
+                tmpl = dep_template.format(name = "{}", target = WHEEL_FILE_PUBLIC_LABEL),
+                select = getattr(native, "select", select),
+            ),
+            visibility = impl_vis,
+        )
+
+    if hasattr(rules, "py_library"):
+        _data_exclude = [
+            "**/* *",
```

Comment:
I seem to recall a recent PR that included names with spaces? Is that being lost here?

Reply:
This is still not merged - #2334.